tools/bridge/bridge/validation_utils.py

```diff
@@ -0,0 +1,73 @@
+from web3 import Web3
+from web3.contract import Contract
+from web3._utils.abi import abi_to_signature
+
+from bridge.contract_abis import MINIMAL_VALIDATOR_PROXY_ABI
+
+
+def validate_contract(contract: Contract) -> None:
+    """Verifies if a given contract exists on the chain.
+
+    It checks if any code is stored for the address. For events and functions
+    in the ABI it will be checked if their signature hash can be found within
+    the code. Throws an exception if the contract could not be verified. Else
+    return nothing.
+    """
+
+    contract_code = contract.web3.eth.getCode(contract.address)
+
+    if not contract_code:
+        raise ValueError(
+            f"The given contract address {contract.address} does not point to a contract!"
+        )
+
+    for description in contract.abi:
+        description_type = description.get("type", "function")
+
+        if description_type in ("constructor", "fallback"):
+            continue
+
+        signature = abi_to_signature(description)
+        signature_hash = Web3.keccak(text=signature)
+
+        if description_type == "function":
+            signature_hash = signature_hash[:4]
+
+        if signature_hash not in contract_code:
+            raise ValueError(
+                f"The contract at the given address {contract.address} does not"
+                f" support the {description_type} with the following signature: {signature}!"
```

Comment:
Black do not break such string automatically. I'll go for it, because it fits into the pattern, that you could just delete the second line without touching the first one. But your are correct, it is not pretty and not worth. I'll put them to the end of the previously line.

Reply:
:heavy_check_mark: